### PR TITLE
Travis docker environment for testing

### PR DESCRIPTION
**Pull request by trws**
Travis has been giving me fits with package installs.  Since I made some docker containers to test raja on linux locally, I thought maybe they would help travis out, DNM.

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
This is sweet 😄

File: .travis.yml [hunk 1]
```diff
@@ -99,7 +80,8 @@ before_install:
   export LD_LIBRARY_PATH=${CUDA_HOME}/lib64:${LD_LIBRARY_PATH} &&
   export PATH=${CUDA_HOME}/bin:${PATH}; fi
 script:
-- ./scripts/travis_build_and_test.sh
+  - docker run --rm --user='root' -v ${TRAVIS_BUILD_DIR}:/usr/src/myapp -w /usr/src/myapp trws/raja-testing:$IMG chown -R raja /usr/src/myapp
```
Comment (davidbeckingsale) [inline .travis.yml]: I only have 1 comment - changing the dir name from `/usr/src/myapp` to something RAJA-y, just to keep the travis logs clear.

**Comment by davidbeckingsale** [comment]
I actually thought of something else... is it worth making a `raja` docker hub user and keeping our images there?

**Comment by trws** [comment]
That would probably make sense, also having the dockerfiles in git.  The 
main remaining blocker is figuring out icc.  Jeff tells me it should be 
kosher as long as the image doesn’t contain a working license, but 
I’m not sure how to accomplish that if it’s generated in the 
install.  Any ideas?

**Comment by DavidPoliakoff** [comment]
Sweet, this'll be great for tutorials as well

**Comment by davidbeckingsale** [comment]
@trws - you can just pass the environment variable (that's securely stored by Travis) into the container right? When you build the image you can just have it use the trial license, then update the config before running our tests.

**Comment by trws** [comment]
That's a good idea, will give that a try and see what happens.

**Comment by davidbeckingsale** [comment]
Hey @trws, this looks great. I think switching over is a no-brainer. My only comment would be the one I raised earlier: should we have a `raja` user on dockerhub for the images?

Once we merge this, I think it would be a good idea to add a section to the README and sphinx docs too.

**Comment by trws** [comment]
Agreed, I've created an organization on dockerhub and will update the dockerfiles and so-forth to refer to it shortly.  If you can send me your dockerhub username I'll add you on as owner as well.

**Comment by trws** [comment]
Ping @rhornung67, review/merge?

**Comment by rhornung67** [comment]
👍 Sorry, merge at will.

**Comment by davidbeckingsale** [comment]
Excellent! Thanks @trws

**Comment by trws** [comment]
Happy to, love having icc testing on my laptop too, seriously.  Also, I'm pretty fond of these time-traveling merges:
image

**Comment by rhornung67** [comment]
delete branch?

**Pull request merged**